### PR TITLE
new geom features

### PR DESCRIPTION
**Pull request by farmakis**
The required changes to support the "new geom features" (#2355) PR in the CloudCompare repo

### CODE REVIEW

File: src/Neighbourhood.cpp [hunk 1]
```diff
@@ -849,6 +849,12 @@ double Neighbourhood::computeFeature(GeomFeature feature)
 		case EigenValue3:
 			value = l3;
 			break;
+		case DegreeOfPlanarity:
+			value = std::log(l1/l3);
```
Comment (dgirardeau) [inline src/Neighbourhood.cpp]: if (std::abs(l3) > std::numeric_limits<double>::epsilon())

File: src/Neighbourhood.cpp [hunk 2]
```diff
@@ -849,6 +849,12 @@ double Neighbourhood::computeFeature(GeomFeature feature)
...
+			break;
+		case DegreeOfLinearity:
+			value = std::log(l1/l2) / std::log(l2/l3);
```
Comment (dgirardeau) [inline src/Neighbourhood.cpp]: if (std::min(std::abs(l2), std::abs(l3)) > std::numeric_limits<double>::epsilon())

File: include/Neighbourhood.h [hunk 3]
```diff
@@ -179,7 +179,10 @@ namespace CCCoreLib
 			Verticality,
 			EigenValue1,
 			EigenValue2,
-			EigenValue3
+			EigenValue3,
+			DegreeOfPlanarity,
+			DegreeOfLinearity
+
```
Comment (dgirardeau) [inline include/Neighbourhood.h]: remove blank line

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: src/Neighbourhood.cpp (hunk 1 above)
Comment (farmakis) [reply src/Neighbourhood.cpp]: Done

**Comment by farmakis** [comment]
The requested changes have been applied

**Review by dgirardeau — APPROVED** [review]

**Comment by dgirardeau** [comment]
Oops, and I realize now that it's when the argument is zero the std::log function will return +inf. Which means that we should also test that l1 is not zero ...

**Comment by farmakis** [comment]
Actually since the eigenvalues are sorted, if I am not mistaken, checking only for l3, automatically implies that l1 and l2 are also valid.
I realized however another case where l2=l3 (perfect sphere), which lead to a division by zero in the formula of DegreeOfLinearity. I am going to apply that too.

**Comment by dgirardeau** [comment]
Will you open a new PR?

**Comment by farmakis** [comment]
> Will you open a new PR?

Yes. im opening it asap
